src/main/scala/RewriteView.scala

```diff
@@ -0,0 +1,109 @@
+package fuselang
+
+/**
+ * AST pass to rewrite views into simple array accesses. Should be used after
+ * type checking.
+ *
+ * For `shrink` views, we rewrite them as:
+ *
+ * view v_a = shrink a[4 * i : 4];
+ * v_a[k];
+ * ==>
+ * ; // remove the view decl
+ * a[4*i + k]
+ *
+ * If `a` itself is a view, we keep rewriting it until we reach a true array.
+ */
+object RewriteView {
+  import Syntax._
+  import CodeGenHelpers._
+
+	// We can use a flat environment because the type checker makes sure things
+  // in the right scope.
+  type T = Map[Id, List[Expr] => Expr]
+
+  def foldExprs(es: List[Expr])(implicit env: T): (List[Expr], T) =
```

Comment:
Having "fold" in the name here is a little weird because this isn't really a fold operation (because it's not higher order). Of course, perhaps having a general `fold` for expressions could be useful?